test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/CustomUriFunctionsTests.cs

```diff
@@ -670,6 +670,34 @@ public void ParseWithCustomUriFunction_AddAsOverloadToBuiltIn()
             }
         }
 
+        [Fact]
+        public void ParseWithCustomFunction_EnumParameter()
+        {
+            try
+            {
+                var enumType = new EdmEnumType("Fully.Qualified.Namespace", "NonFlagShape", EdmPrimitiveTypeKind.SByte, false);
+                enumType.AddMember("Rectangle", new EdmEnumMemberValue(1));
+                enumType.AddMember("Triangle", new EdmEnumMemberValue(2));
+                enumType.AddMember("foursquare", new EdmEnumMemberValue(3));
+                var enumTypeRef = new EdmEnumTypeReference(enumType, false);
+
+                FunctionSignatureWithReturnType signature =
+                    new FunctionSignatureWithReturnType(EdmCoreModel.Instance.GetBoolean(false), enumTypeRef);
+
+                CustomUriFunctions.AddCustomUriFunction("enumFunc", signature);
+
+                var fullUri = new Uri("http://www.odata.com/OData/People" + "?$filter=enumFunc('Rectangle')");
+                ODataUriParser parser = new ODataUriParser(HardCodedTestModel.TestModel, new Uri("http://www.odata.com/OData/"), fullUri);
+
+                var enumFuncWithArgs = parser.ParseFilter().Expression.ShouldBeSingleValueFunctionCallQueryNode("enumFunc").And.Parameters.ToList();
+                enumFuncWithArgs[0].ShouldBeEnumNode(new ODataEnumValue("Rectangle", enumType.FullName));
+            }
+            finally
+            {
+                CustomUriFunctions.RemoveCustomUriFunction("enumFunc").Should().BeTrue();

```

Comment:
>CustomUriFunctions [](start = 16, length = 18)

Why should we use "try{} finally{}"?

Reply:
CustomUriFunctions is static class. We need to add  and then remove function registration. try/finally allows to ensure that registration will be removed even if test fails and other test will not be affected.

It's the pattern in CustomUriFunctionsTests